tests/test_main.py

```diff
@@ -130,108 +131,401 @@ def _run_step(self, runtime_step_config):
-
-def test_no_arguments():
-    _run_main_test(None, 2)
-
-def test_help():
-    _run_main_test(['--help'], 0)
-
-def test_bad_arg():
-    _run_main_test(['--bad-arg'], 2)
-
-def test_config_file_does_not_exist():
-    _run_main_test(['--step', 'generate-metadata', '--config-file', 'does-not-exist.yml'], 101)
-
-def test_config_file_not_json_or_yaml():
-    _run_main_test(['--step', 'generate-metadata'], 102,
-        ": blarg this: is {} bad syntax")
-
-def test_config_file_no_root_tssc_config_key():
-    _run_main_test(['--step', 'generate-metadata'], 103, "{}")
-
-def test_config_file_valid_yaml():
-    TSSCFactory.register_step_implementer(FooStepImplementer, True)
-    _run_main_test(['--step', 'foo'], None,
-        '''tssc-config: {}'''
-    )
-
-def test_config_file_valid_json():
-    TSSCFactory.register_step_implementer(FooStepImplementer, True)
-    _run_main_test(['--step', 'foo'], None,
-        '''{"tssc-config":{}}'''
-    )
-
-def test_required_step_config_missing():
-    TSSCFactory.register_step_implementer(RequiredStepConfigStepImplementer, True)
-    _run_main_test(['--step', 'required-step-config-test'], 200,
-        '''{"tssc-config":{}}'''
-    )
-
-def test_required_step_config_pass_via_config_file():
-    TSSCFactory.register_step_implementer(RequiredStepConfigStepImplementer, True)
-    _run_main_test(['--step', 'required-step-config-test'], None,
-        '''---
-        tssc-config:
-          required-step-config-test:
-            implementer: RequiredStepConfigStepImplementer
-            config:
-              required-config-key: "hello world"
-        '''
-    )
-
-def test_required_step_config_pass_via_runtime_arg_missing():
-    TSSCFactory.register_step_implementer(RequiredRuntimeStepConfigStepImplementer, True)
-    _run_main_test(
-        [
-            '--step', 'required-runtime-step-config-test',
-            '--step-config', 'wrong-config="hello world"'
-        ],
-        200,
-        '''---
-        tssc-config: {}
-        '''
-    )
-
-def test_required_step_config_pass_via_runtime_arg_valid():
-    TSSCFactory.register_step_implementer(RequiredRuntimeStepConfigStepImplementer, True)
-    _run_main_test(
-        [
-            '--step', 'required-runtime-step-config-test',
-            '--step-config', 'required-rutnime-config-key="hello world"'
-        ],
-        None,
-        '''---
-        tssc-config: {}
-        '''
-    )
+                
+                if expected_results:
+                    with open(os.path.join(results_dir_path, "tssc-results.yml"), 'r') as step_results_file:
+                        results = yaml.safe_load(step_results_file.read())
+                        self.assertEqual(results, expected_results)
+    
+    def test_init(self):
+        """
+        Notes
+        -----
+        See https://medium.com/opsops/how-to-test-if-name-main-1928367290cb
+        """
+        from tssc import __main__
+        with patch.object(__main__, "main", return_value=42):
```

Comment:
is there a way to avoid having patches 3 deep?

Reply:
that test was already there, only change to it was to use the unitest.mock patch rather then mock.patch which it was using.

the test is a bit of hack to get 100% test coverage on 1 line of code which does the boot strapping of the init script. as described in the link in the comment.

If we wnat to change this test in some functional way, as my friend @kfrankli would say, that should be another MR.

Reply:
Nah, I got you, it was just something I noticed. Better have this than a cov ignore force.